base/loading.jl

```diff
@@ -2468,6 +2614,16 @@ end
             end
             for chi in includes
                 f, ftime_req = chi.filename, chi.mtime
+                if !isfile(f)
```

Comment:
I would very much like to introduce a notion of "depot-local" files here.  Here's how I see it working:

If we have a package `Foo.jl` which has been precompiled, that cache should reference the source files as something like: `@depot/packages/Foo_jll/a2b2c3/src/Foo.jl`.  During cache header checking, we string-replace `^@depot` to the same depot that the cache file is currently being loaded from.  I do not think it's worthwhile to allow this path to resolve to other depots on the depot path; it's too easy to find the "wrong" files, and I think it's reasonable to expect an atomic move of all associated resources for a package image.

When serializing out dependency paths, we simply check if the depot we're serializing into is a superpath of the dependency path.  If it is, we emit `@depot/...`, otherwise we emit the full path.

To be clear, I'm okay with this being pushed off for future work :)

Reply:
Maybe open this as an issue?

Reply:
https://github.com/JuliaLang/julia/issues/47943